build/mvn

```diff
@@ -25,7 +25,7 @@ _CALLING_DIR="$(pwd)"
 _COMPILE_JVM_OPTS="-Xms2g -Xmx2g -XX:ReservedCodeCacheSize=1g -Xss128m"
 
 if [ "$CI" ]; then
-  export MAVEN_CLI_OPTS="--no-transfer-progress --errors --fail-fast"
+  export MAVEN_CLI_OPTS="--no-transfer-progress --errors --fail-fast --settings ${GITHUB_WORKSPACE}/.github/maven/settings.xml"
```

Comment:
```
-Dmaven.resolver.transport=wagon
```
https://maven.apache.org/guides/mini/guide-resolver-transport.html#how-to-upgrade-from-wagon-or-%E2%80%9Cnative-transport-does-not-work%E2%80%9D

Reply:
It seems Wagon also requires settings.xml if we need to set connection timeout.
https://maven.apache.org/guides/mini/guide-http-settings.html#connection-timeouts

Reply:
The default timeout of `wagon` also changed?

Reply:
<img width="596" alt="image" src="https://user-images.githubusercontent.com/1475305/231379851-6d6be5ae-429f-4040-863a-d9ad5d885d2c.png">

@bowenliang123 As you said, `Wagon's default connect timeout is 60000ms (1min)`, Do we still need to increase it?

Reply:
Not sure about this. @LuciferYang 
And according to @cxzl25 's advice, we are trying to switch to use Wagon in #4699 .